src/main/java/core/basesyntax/shop/service/impl/FruitShopServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.shop.service.impl;
+
+import core.basesyntax.shop.dao.FruitShopDao;
+import core.basesyntax.shop.dao.FruitShopDaoImpl;
+import core.basesyntax.shop.dao.InsufficientGoodsException;
+import core.basesyntax.shop.model.Fruit;
+import core.basesyntax.shop.service.FruitShopService;
+import java.util.Map;
+
+public class FruitShopServiceImpl implements FruitShopService {
```

Comment:
Even if you use that interesting way to make strategy, try to unite them (here must be 2 not 4).

Reply:
we talked about possible change in different operations logic. that's why i keep 4 operation. in the name of SOLID!